client/state/sharing/publicize/actions.js

```diff
@@ -71,6 +75,44 @@ export function fetchConnections( siteId ) {
 }
 
 /**
+ * Triggers a network request to request a Publicize connection for the
+ * specified site ID.
+ *
+ * @param  {Number} siteId       Site ID
+ * @param  {Number} connectionId ID of the connection to be fetched.
+ * @param  {String} label        Name of the service to be fetched.
+ * @return {Function}            Action thunk
+ */
+export function fetchConnection( siteId, { ID: connectionId, label } ) {
```

Comment:
> Right, so the original purpose was to refresh an existing connection's data.

So I think @aduth's suggested signature is a easier to understand and reuse. Here the caller will need to do a tiny bit more work from your use case eg:
```javascript
const { id: connectionId } = connection;
fetchConnection( siteId, connectionId )
```

> The label would allow us to display a service-agnostic error message

Do we not get a label from the error response?